apps/forwarding/models.py

```diff
@@ -0,0 +1,177 @@
+from reversion.models import Version
+
+from apps.commcare.models import BaseModel
+from apps.exports.templatetags.dateformat_tags import readable_timedelta
+
+
+class ForwardingDatabase(BaseModel):
+    """Database where queries are executed (source)."""
+
+    name = models.CharField(max_length=100)
+    connection_string = models.CharField(max_length=500)
+    owner = models.ForeignKey(
+        settings.AUTH_USER_MODEL, on_delete=models.CASCADE
+    )
+
+    def __str__(self):
+        return self.name
+
+
+class ForwardingDestination(BaseModel):
+    """API endpoint where query results are forwarded (destination)."""
+
+    name = models.CharField(max_length=100)
+    api_url = models.URLField(
+        max_length=500, help_text=_('API endpoint URL where data will be sent')
+    )
+    api_username = models.CharField(
+        max_length=100,
+        blank=True,
+        help_text=_('Username for basic authentication'),
+    )
+    api_password = models.CharField(
+        max_length=200,
+        blank=True,
+        help_text=_('Password for basic authentication'),
+    )
+    owner = models.ForeignKey(
+        settings.AUTH_USER_MODEL, on_delete=models.CASCADE
+    )
+
+    def __str__(self):
+        return self.name
+
+
+@reversion.register()
+class ForwardingConfig(BaseModel):
+    """Configuration for a data forwarding job."""
+
+    name = models.CharField(max_length=100)
+    database = models.ForeignKey(ForwardingDatabase, on_delete=models.CASCADE)
+    destination = models.ForeignKey(
+        ForwardingDestination, on_delete=models.CASCADE
+    )
+    query = models.TextField(
+        help_text=_(
+            'SQL query to execute. Must return exactly one row with one field '
+            'containing the payload (typically JSON).'
+        )
+    )
+    query_params = models.TextField(
+        blank=True,
+        help_text=_(
+            'Query parameters (one per line). Mapped to :param1, :param2, '
+            'etc. in the query.'
+        ),
+    )
+    created_by = models.ForeignKey(
+        settings.AUTH_USER_MODEL, on_delete=models.CASCADE
+    )
+    schedule = models.ForeignKey(
+        'schedules.Schedule',
+        on_delete=models.SET_NULL,
+        null=True,
+        blank=True,
+        help_text=_('Scheduling configuration for automatic forwarding.'),
+    )
+    is_paused = models.BooleanField(
+        default=False,
+        help_text=_(
+            'Pausing will disable automatic forwarding. You can still '
+            'manually run it.'
+        ),
+    )
+
+    def __str__(self):
+        return self.name
+
+    @property
+    def last_run(self):
+        runs = self.runs.exclude(status=ForwardingRun.Status.QUEUED)
+        return runs.order_by('-created_at')[0] if runs.exists() else None
+
+    def has_queued_runs(self):
+        if self.runs.exists():
+            return (
+                self.runs.order_by('-created_at')[0].status
+                == ForwardingRun.Status.QUEUED
```

Comment:
Is it ever possible for the last run to be some other status but an earlier run to still be queued?